VMEncryption/main/patch/redhatPatching.py

```diff
@@ -109,7 +101,16 @@ def install_extras(self):
             packages.remove('util-linux')
 
         if self.command_executor.Execute("rpm -q " + " ".join(packages)):
-            self.command_executor.Execute("yum install -y " + " ".join(packages))
+            ret_val = self.command_executor.Execute("yum install -y " + " ".join(packages))
```

Comment:
but the rpm -q is there for that no?

Reply:
Yes but the rpm -q call is checking against an entire list of packages.  Some of the packages in that list might already be on the system, and some might not.  If it finds even one missing package, it will drop into yum install -y which will then try to install the entire list of packages.  If one of those packages was already installed yum install -y will return a failure code even after installing the missing packages because a subset of the packages it was asked to install were already there.

Reply:
Interesting.
Let me add a second rpm -q then to handle that scenario.